### PR TITLE
Support for list at least for json

### PR DESCRIPTION
**Pull request by Judimax**

### CODE REVIEW

**Comment by ilteoood** [comment]
Could you add some tests that proved that what you did is correct?

**Comment by Judimax** [comment]
Greetings I have made the edits as requested I updated the android/app/build.gradle to version 34 to get the andrioid integrations test to pass not sure how that fares with backwards comptability

**Comment by Judimax** [comment]
the changes
* [UPDATE] Updated _decodeFromMap in simple_translator.dart to handle subMap as a dynamic type, reintroducing checks and handling for subMap being a List.
* [UPDATE] Modified calculateSubmap in simple_translator.dart to return a dynamic type and reintroduced logic to handle the decodedSubMap as either Map or List, including handling list indexing and nested data structures.

File: lib/utils/simple_translator.dart [hunk 1]
```diff
@@ -47,25 +47,54 @@ class SimpleTranslator {
...
-    final String lastKeyPart = key.split(this.keySeparator!).last;
-    final result = subMap[lastKeyPart] is String ? subMap[lastKeyPart] : null;
+    final dynamic subMap = calculateSubmap(key);
+    if (subMap is List) {
+      final String lastKeyPart = key.split(this.keySeparator!).last;
+      if (int.parse(lastKeyPart) >= subMap.length) {
+        missingKeyTranslationHandler!(key);
+        return null;
+      }
+      final result = subMap[int.parse(lastKeyPart)] is String
```
Comment (ilteoood) [inline lib/utils/simple_translator.dart:57]: subMap[int.parse(lastKeyPart)] is calculated multiple times. Extract its value in a variable

File: lib/utils/simple_translator.dart [hunk 2]
```diff
@@ -47,25 +47,54 @@ class SimpleTranslator {
   }
 
   String? _decodeFromMap(final String key) {
-    final Map<dynamic, dynamic> subMap = calculateSubmap(key);
-    final String lastKeyPart = key.split(this.keySeparator!).last;
-    final result = subMap[lastKeyPart] is String ? subMap[lastKeyPart] : null;
+    final dynamic subMap = calculateSubmap(key);
+    if (subMap is List) {
+      final String lastKeyPart = key.split(this.keySeparator!).last;
+      if (int.parse(lastKeyPart) >= subMap.length) {
```
Comment (ilteoood) [inline lib/utils/simple_translator.dart:53]: int.parse(lastKeyPart) is done multiple times. extract the result in a variable

File: lib/utils/simple_translator.dart [hunk 3]
```diff
@@ -47,25 +47,54 @@ class SimpleTranslator {
...
-    if (result == null && key.length > 0) {
-      missingKeyTranslationHandler!(key);
-    }
+      if (result == null && key.length > 0) {
+        missingKeyTranslationHandler!(key);
+      }
+
+      return result;
+    } else if (subMap is Map) {
+      final String lastKeyPart = key.split(this.keySeparator!).last;
```
Comment (ilteoood) [inline lib/utils/simple_translator.dart:67]: lastKeyPart is calculated in the exact same way for both list and map. extract its value outside the if statement.

File: lib/utils/simple_translator.dart [hunk 4]
```diff
@@ -47,25 +47,54 @@ class SimpleTranslator {
   }
 
   String? _decodeFromMap(final String key) {
-    final Map<dynamic, dynamic> subMap = calculateSubmap(key);
-    final String lastKeyPart = key.split(this.keySeparator!).last;
-    final result = subMap[lastKeyPart] is String ? subMap[lastKeyPart] : null;
+    final dynamic subMap = calculateSubmap(key);
```
Comment (ilteoood) [inline lib/utils/simple_translator.dart:50]: As it's not only a map anymore, the name `subMap` is not correct anymore

File: lib/utils/simple_translator.dart [hunk 5]
```diff
@@ -47,25 +47,54 @@ class SimpleTranslator {
...
-    Map<dynamic, dynamic>? decodedSubMap = decodedMap;
+    dynamic decodedSubMap = decodedMap;
     translationKeySplitted.forEach((listKey) {
-      final subMap = (decodedSubMap ?? Map())[listKey];
-      decodedSubMap = subMap is Map ? subMap : Map();
+      decodedSubMap =
+          decodedSubMap is Map || decodedSubMap is List ? decodedSubMap : Map();
+      dynamic subMap;
+      if (decodedSubMap is List) {
+        if(int.parse(listKey) < decodedSubMap.length){
```
Comment (ilteoood) [inline lib/utils/simple_translator.dart:88]: int parse repeated multiple times. extract it.

File: lib/utils/simple_translator.dart [hunk 6]
```diff
@@ -47,25 +47,54 @@ class SimpleTranslator {
...
+          subMap = (decodedSubMap ?? List.empty())[int.parse(listKey)];
+        }
+        else {
+          subMap = Map();
+        }
+      } else if( decodedSubMap is Map) {
```
Comment (ilteoood) [inline lib/utils/simple_translator.dart:94]:
```suggestion
      } else if (decodedSubMap is Map) {
```

**Review by ilteoood — CHANGES_REQUESTED** [review]

**Comment by Judimax** [comment]
I appreciate the lazer focus on best coding practices I haven't seen devotion like that in my day to day so I had to dumb it back

**Comment by Judimax** [comment]
Checking in any updates

**Comment by ilteoood** [comment]
Hi @Judimax, sorry for being late in the review process.

I don't think that this PR is something that should be merged atm: it complicates the implementation and what you want desire can be achieved using a syntax like:
```
  'key': {
    '0': 'value'
    ...
  }
```

**Pull request closed without merging**